lib/Parse/ParseDecl.cpp

```diff
@@ -1460,6 +1347,41 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
         .fixItReplace(Tok.getLoc(), "available");
   }
 
+
+  if (DK == DAK_Count && Tok.getText() == "warn_unused_result") {
```

Comment:
It requires having a switch case for it, but you can throw out everything else the same way you do here. I do see that it's annoying in the rest of the compiler; maybe we should have a new case in Attr.def for removed attributes.